main.js

```diff
@@ -0,0 +1,276 @@
+function getRandomInt(max) {
+  return Math.floor(Math.random() * Math.floor(max));
+}
+
+//how to win..
+// winning combos that are possible (array of arrays)
+let winningCombinations = [
+  ["cell-0", "cell-1", "cell-2"],
+  ["cell-3", "cell-4", "cell-5"],
+  ["cell-6", "cell-7", "cell-8"],
+  ["cell-2", "cell-5", "cell-8"],
+  ["cell-0", "cell-3", "cell-6"],
+  ["cell-1", "cell-4", "cell-7"],
+  ["cell-0", "cell-4", "cell-8"],
+  ["cell-2", "cell-4", "cell-6"],
+];
+
+// function to determine if someone has won
+function hasWon(moves, winningCombinations) {
+  // create new array...
+  let foundResults = winningCombinations.filter(
+    // look at arrays inside winning combos array..
+    (array) =>
+      // for each small array look to see if each item is present.. if it is add it to the new array
+      array.filter((item) => {
+        // return what index in the players moves the winning combo item is.. if it exists in player moves, add to new array
+        return moves.indexOf(item) > -1;
+        // when length is three add this array to newly created array
+      }).length === 3
+  );
+
+  // if foundResults includes an array (1) of length three...
+  if (foundResults.length > 0) {
+    // announce someone won, so that comPlay isn't called
+    winStatus = true;
+    // change border color item.setAttribute("id", "winningCell"); so appropriate css is applied
+    foundResults[0].forEach((item) => {
+      tempElement = document.getElementById(item);
+      console.log(tempElement);
+      tempElement.id = "winningCell";
+    });
+
+    //alerts who has won if somebody wins
+    if (systemStatus === false) {
+      status.innerText = `${userName} has won!`;
+    } else if (systemStatus === true && oStatus === "computer") {
+      status.innerText = "The computer has won";
+    } else {
+      status.innerText = `${userOName} won`;
+    }
+
+    // disable all play board buttons so play cant continue
+    for (cell of cellButton) {
+      cell.disabled = true;
+    }
+
+    // enable replay button so play can start over..
+    replay.disabled = false;
+    // stop the timer
+    clearInterval(playTimer);
+
+    // DRAW SCENARIO
+    // if XMoves + OMoves = 9 then it is a draw and nobody wins (only triggered if nobody has won yet)
+  } else if (XMoves.length + OMoves.length === 9) {
+    status.innerText = "it's a draw";
+    let drawArray = XMoves.concat(OMoves); //combine all moves back into one array
+    drawArray.forEach((item) => {
+      //make all cells have rainbow border css
+      tempElement = document.getElementById(item);
+      console.log(tempElement);
+      tempElement.id = "winningCell";
+    });
+
+    // disable all play board buttons so play cant continue
+    for (cell of cellButton) {
+      cell.disabled = true;
+    }
+
+    // enable replay button so play can resume..
+    replay.disabled = false;
+
+    // stop the timer
+    clearInterval(playTimer);
+  }
+}
```

Comment:
Very interesting use of .length to take care of a lot of the logic for win and draw scenarios. This really let you get away with far less overall code while still completing all the different necessary logic checks!